packit_service/worker/build/build_helper.py

```diff
@@ -91,6 +96,35 @@ def local_project(self) -> LocalProject:
             )
         return self._local_project
 
+    @property
+    def pr_id(self) -> int:
+        if self._pr_id is None:
+            self._pr_id = int(self.metadata.event_dict["identifier"])
+        return self._pr_id
+
+    @property
+    def is_reporting_allowed(self) -> bool:
+        username = self.project.service.user.get_username()
+        if self._is_reporting_allowed is None:
+            self._is_reporting_allowed = self.base_project.can_merge_pr(username)
+        return self._is_reporting_allowed
+
+    @property
+    def base_project(self) -> GitProject:
+        """
+        Getting the source project info from PR,
+        In case of build events we loose the source info.
+        """
+        if self._base_project is None:
+            self._base_project = self.project.get_pr(pr_id=self.pr_id).source_project
```

Comment:
This will not work for non-pr cases. We should return `base_project` in that case.

Reply:
The best I could think of was by cathing the value error here which will come from pr_id, in case of non-pr events.